hedera-node/hedera-network-admin-service-impl/src/main/java/com/hedera/node/app/service/networkadmin/impl/config/SemanticVersionConfig.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.node.app.service.networkadmin.impl.config;
+
+import com.hedera.hapi.node.base.SemanticVersion;
+import com.swirlds.config.api.ConfigData;
+import com.swirlds.config.api.ConfigProperty;
+
+@ConfigData("version")
+public record SemanticVersionConfig(
+        @ConfigProperty(defaultValue = "0.39.0") SemanticVersion servicesVersion,
+        @ConfigProperty(defaultValue = "0.39.0") SemanticVersion hapiVersion) {}
```

Comment:
In my variant of this class, I defined the "servicesVersion" field to be called "services" and the hapiVersion to be "hapi". This was so in the config it would be "version.services" and "version.hapi" instead of "version.servicesVersion" and "version.hapiVersion" since this seemed redundant.

I agonized a bit over it though, because the existing fields are something more like "proto.version" and "services.version", implying they might be part of different config objects (with a @ConfigData of services or a @ConfigData of hapi).

I also agonized over what, if any, default version to give them. Ideally they would have no default version, so that if a build somehow failed to get a proper version, then the node would simply fail to start. Getting the version number right is essential for selecting which schema to use. If we do give it a default, I recommend we give it a default version of 0, which is the genesis version. If the node software starts but with no version information, then it will fail to start unless it is also a genesis case. If you tried to start the node software where the existing state had a higher version number, it would fail to start. So it seems like the safest default would be "0.0.0"

Reply:
A default is required in order for unit tests to work. I changed the default to "0.0.0" and I changed to use the naming you suggest.